### ISSUE TITLE
Time Scretch on Raspberry Pi 0.28 build on Raspian

### ISSUE THREAD

**Issue by alex-nemirovsky** [bennettpeter/mythtv#3]
Hi All,

Does time stretch work for anyone on Raspberry Pi? I’m using RPv3.  
The audio and video is all jittery for me on anything except 1.0x.   < 1.0 or > 1.0.  Doesn’t seem to matter. 

Tried both OpenMax profile and OpenGL profile and the problem is seen in both.
CPU load is 25% on all 4 cores during normal playback and about 40% with time stretch enabled.

Anyone else having issues? Known problem?

I’m using:
-build mythtv-light_0.28-32-g6eff083-0_armhf_jessie.deb for Mythtv 0.28 running
on latest version of Raspian. 
- OpenMax video profile, Video memory set for 256MB video ram, 4 CPU. Playing 1080i and 720p ASTC over-the-air content recorded by HDHR3.  Mpeg2 license installed an verified.
  -Screen resolution set for 720p to 720p HDTV.
- ALSA default audio forced to Stereo PCM over HDMI to the HDTV.
  -Wired Ethernet.  

Plays fine on 0.28 x86 front end running 14.04 mythbuntu from the same backend (also 14.04 with Myth 0.28).

**Comment by bennettpeter** [comment]
Time stretch does not currently work on raspberry pi.

**Comment by alex-nemirovsky** [comment]
Thanks for the response Peter.  Do you have any other info you can share about this? 
Such as:
- Do we know if it a bug or a due to a major component being missing? If so, where or what component
- Do we know if anyone is already working on this or planning to work on it soon? If so, who?

**Comment by bennettpeter** [comment]
I believe this is caused by the raspberry pi not having enough processing power. It can probably be fixed by changing the way things are done. My guess is the stretching or squeezing of the audio is a cpu-intensive process. 

Nobody is working on this. 

It seems to me that time-stretch is not a feature that would be widely used.

If this is important to you please open a trac ticket (https://code.mythtv.org/trac/newticket) and set component as "Ports - rPi". That will be assigned to me.

**Comment by alex-nemirovsky** [comment]
I plugged in a USB Audio adapter (Griffin iMic) and time stretch is almost useable.  Still not smooth but a lot better than the audio coming out of the HDMI output.

I'm not sure its a CPU load issue completely.   I set OpenMAX to single CPU and it improves slightly and video still plays fine when time-stretch is not enable.  Plenty of CPU for playback for the 1080i and 720p content I am using in my control case.

Without time stretch using onboard audio playing 1080i:
09:24:24 PM  CPU    %usr   %nice    %sys %iowait    %irq   %soft  %steal  %guest  %gnice   %idle
09:24:34 PM  all    3.60    0.00    2.78    0.00    0.00    0.67    0.00    0.00    0.00   92.95
09:24:34 PM    0    3.27    0.00    0.98    0.00    0.00    2.83    0.00    0.00    0.00   92.92
09:24:34 PM    1    2.42    0.00    3.43    0.00    0.00    0.00    0.00    0.00    0.00   94.15
09:24:34 PM    2    4.33    0.00    2.32    0.00    0.00    0.00    0.00    0.00    0.00   93.35
09:24:34 PM    3    4.37    0.00    4.27    0.00    0.00    0.00    0.00    0.00    0.00   91.35

Time stretch enabled using onboard audio playing same 1080i content:
09:23:24 PM  CPU    %usr   %nice    %sys %iowait    %irq   %soft  %steal  %guest  %gnice   %idle
09:23:34 PM  all   19.67    0.00    1.48    0.03    0.00    0.46    0.00    0.00    0.00   78.37
09:23:34 PM    0   14.76    0.00    1.15    0.00    0.00    1.88    0.00    0.00    0.00   82.20
09:23:34 PM    1   37.26    0.00    1.81    0.00    0.00    0.00    0.00    0.00    0.00   60.93
09:23:34 PM    2   23.80    0.00    0.30    0.00    0.00    0.00    0.00    0.00    0.00   75.90
09:23:34 PM    3    2.45    0.00    2.65    0.10    0.00    0.00    0.00    0.00    0.00   94.80

Not sure why last CPU is not loading like the others.  I'll file ticket.  thanks
